### PR TITLE
Add documentation for smart notifications

### PR DESCRIPTION
**Pull request by nquinquenel**

### CODE REVIEW

File: spec/smart_notifications.adoc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+= Smart notifications
+
+== Overview
+
+Smart notifications, sometimes called dev notifications, allow developers using xref:glossary.adoc#connected_mode[Connected Mode] in SonarLint to receive in-IDE notifications from SonarQube or SonarCloud when:
+
+* the Quality Gate status (failed/success) of a project open in the IDE changes
+* a SonarQube or SonarCloud analysis raises new xref:glossary.adoc#finding[findings] introduced by this developer in a project open in the IDE
+
+SonarLint polls the server (SQ/SC) every minute to retrieve any new notifications up to 1 day old. When the polling has been done, the timestamp of the polling date is stored and will be used next time as a reference, so the client cannot retrieve the same notification later.
```
Comment (damien-urruty-sonarsource) [inline spec/smart_notifications.adoc:11]: I know that @henryju does not like the term "server". We probably need to find a term for this concept

**Review by damien-urruty-sonarsource — APPROVED** [review]
Good job, LGTM :+1:

File: spec/smart_notifications.adoc (hunk 1 above)
Comment (henryju) [reply spec/smart_notifications.adoc:11]: maybe simply "SonarLint will poll SonarQube or SonarCloud"

**Review by henryju — APPROVED** [review]
Very good! Thanks.

**Pull request merged**